### PR TITLE
[Feature] Exam Archival

### PR DESCRIPTION
**Pull request by ivanchimeno**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

It should be possible for instructors to archive/export an exam after it is over. Archiving an exam compresses all student submissions for each exam exercise into a single zip file (similar to the existing course archival functionality).

### Description

This PR implements the archival of a specific exam. It does this by:

1. Extending the existing export functionalities in `CourseExportService` to support exporting specific exams. The service is renamed to `CourseExamExportService` since they share the same implementation.
2. Adding to new endpoints in `ExamResource` that enable instructors to archive an exam and download it.
3. Adds a new Angular component that implements the archive and download buttons for courses and exams
4. Implements the archive functionality in the exam details component

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create an exam with Text, File Upload, Modeling, and Programming exercises
4. Participate in the exam with a couple of users by generating submissions
5. During the exam and as an instructor, navigate to Course Administration -> Exams -> the exam
6. Make sure the archive button isn't visible
7. When the exam finishes, you should be able to archive the exam. Click on the button and wait until it's done
8. A download button should appear and you should be able to download the zip file.
9. Make sure the zip files contains all student submissions for each exercise
10. You can also log in as a teaching assistant and make sure that you don't have access to the archive button (only instructors and admins have access to it)

### Test Coverage

CourseExamExportService.java 80%
CourseService.java 93%
GroupNotificationService.java 70%
ExamService.java 93%
ExamResource.java 90%
course-detail.component.ts 67%
exam-management.service.ts 66%
exam-detail.component.ts 100%
course-exam-archive-button.component.ts 90%

### Screenshots

The archive button appears when the exam is over:
image

A warning is displayed when pressing the archive button:
image

It's possible to download the zip file once archving is done:
image

### CODE REVIEW

**Review by katjanakosic — DISMISSED** [review]
Tested on ts2. Works as expected. But the button "Archive Exam" seems a little bit off.

image

**Review by airelawaleria — APPROVED** [review]
Tested on ts2. with @katjanakosic @marvinchimeka . Workd as anticipated
Same issue with button

**Review by iozbeyli — COMMENTED** [review]
I added a couple of improvement suggestions.

File: src/main/webapp/app/shared/components/course-exam-archive-button/course-exam-archive-button.component.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        this.archiveButtonText = exportState === 'RUNNING' ? progress : this.getArchiveButtonText();
+
+        if (exportState === 'COMPLETED') {
+            this.jhiAlertService.success(this.getArchiveSuccessText());
+            this.reloadCourseOrExam();
+        }
+    }
+
+    reloadCourseOrExam() {
+        if (this.archiveMode === 'Exam' && this.exam) {
```
[the remark this answers is not in the document]
Comment (ivanchimeno) [reply src/main/webapp/app/shared/components/course-exam-archive-button/course-exam-archive-button.component.ts:104]: I think it's ok :D

**Pull request merged**